xonsh/tools.py

```diff
@@ -81,14 +81,33 @@ def __init__(self, returncode, command, output=None, stderr=None,
         self.completed_command = completed_command
 
 
-def expandpath(path):
+def expand_path(s, expand_user=True):
```

Comment:
Ideas on adding test cases(for expand_user part): like `p = expand_path('~/foo/bar.txt'); assert p.endswith('/foo/bar.txt') and not p.startswith('~')`.